### PR TITLE
fix: release process fails due to timeout from bytecode issues with arm64

### PR DESCRIPTION
**Pull request by awsluja**
#### Description of changes
Currently, building our binaries for arm64 adds a significant amount of time to our release process, largely due to errors with generating bytecode that occur on every file in node_modules. This process typically takes about 35 minutes to run, but is currently taking 1h 19m due to a recent increase in our node_modules folder. The additional time has led to failures with publishing our binaries to S3 due to a 1h timeout for our aws cli login.

image

This change reduces the time it takes to build our CLI binaries on release,release_rc,run-e2e-with-rc, & tagged-release branches by about an hour, by eliminating the bytecode errors. As a result, our CLI binaries can successfully publish to S3 again.
image

Resulting files in S3:
image

CI Run:
https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/11594/workflows/94d829a6-a835-453a-813d-b1b141a4e768

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .circleci/local_publish_helpers.sh [hunk 1]
```diff
@@ -71,7 +71,22 @@ function generatePkgCli {
   # Build pkg cli
   cp package.json ../build/node_modules/package.json
   if [[ "$CIRCLE_BRANCH" == "release" ]] || [[ "$CIRCLE_BRANCH" =~ ^run-e2e-with-rc\/.* ]] || [[ "$CIRCLE_BRANCH" =~ ^release_rc\/.* ]] || [[ "$CIRCLE_BRANCH" =~ ^tagged-release ]]; then
-    npx pkg -t node14-macos-x64,node14-linux-x64,node14-linux-arm64,node14-win-x64 ../build/node_modules --out-path ../out
+    # This will generate a file 'amplify-pkg' in the '../tmp-out' directory for our arm64 binary.
+    # The file name does not include the generated <platform>-<arch> suffixes because there is a single target.
+    npx pkg --no-bytecode -t node14-linux-arm64 ../build/node_modules --out-path ../tmp-out
```
Comment (sobolk) [inline .circleci/local_publish_helpers.sh:76]: Could you try `-o` aka `--output` parameter ti give it proper name and avoid extra tmp dir ?

see https://www.npmjs.com/package/pkg .

File: .circleci/local_publish_helpers.sh [hunk 2]
```diff
@@ -71,7 +71,22 @@ function generatePkgCli {
...
+    # platforms (macos/linux/win) being built in the same pkg call.
+    npx pkg -t node14-macos-x64,node14-linux-x64,node14-win-x64 ../build/node_modules --out-path ../out
+
+    # this renames the x64 files so that they have the appropriate suffix 'x64'
+    mv ../out/amplify-pkg-macos ../out/amplify-pkg-macos-x64
+    mv ../out/amplify-pkg-linux ../out/amplify-pkg-linux-x64
+    mv ../out/amplify-pkg-win.exe ../out/amplify-pkg-win-x64.exe
+
+    # this renames the arm64 file so that it has the appropriate suffix 'linux-arm64'
+    mv ../tmp-out/amplify-pkg ../out/amplify-pkg-linux-arm64
```
Comment (sobolk) [inline .circleci/local_publish_helpers.sh:89]: maybe we should just call npx pkg 4 times and use `-o` aka `--output` parameter ?
i.e.
`npx pkg -t node14-macos-x64 ../build/node_modules --out-path ../out --output amplify-pkg-macos-x64` and so on ?

Comment (awsluja) [reply .circleci/local_publish_helpers.sh:89]: will experiment with --o; I had issues with it misinterpreting ../out/amplify-pkg-linux-x64 as a directory but there were other issues I was working through as well. let me try again

Comment (awsluja) [reply .circleci/local_publish_helpers.sh:89]: running each one individually adds about 5 minutes: 

current (11 min) https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/11589/workflows/b3912cf6-9c09-47aa-a6b6-ab844920c071/jobs/412533

after (16 min) https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/11594/workflows/94d829a6-a835-453a-813d-b1b141a4e768/jobs/412551

Comment (awsluja) [reply .circleci/local_publish_helpers.sh]: I am ok to be slower here, for readability; this will be replaced soon anyway. PR updated!

**Review by edwardfoyle — APPROVED** [review]

**Review by evcodes — APPROVED** [review]

File: .circleci/local_publish_helpers.sh (hunk 2 above)
Comment (sobolk) [reply .circleci/local_publish_helpers.sh]: This might help https://stackoverflow.com/questions/3004811/how-do-you-run-multiple-programs-in-parallel-from-a-bash-script .

Comment (sobolk) [reply .circleci/local_publish_helpers.sh]: yet one more option to explore is that `-o` seems to accept name template, so there should be a way to force it to include arch component.
